src/memote_webservice/schemas.py

```diff
@@ -0,0 +1,32 @@
+# Copyright (c) 2018, Novo Nordisk Foundation Center for Biosustainability,
+# Technical University of Denmark.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from marshmallow import Schema, fields
+
+
+class SubmitRequest(Schema):
+    model = fields.Field(description="Metabolic model file", required=True)
+
+    class Meta:
+        strict = True
+
+
+class SubmitResponse(Schema):
```

Comment:
We have to using `strict` in most places. Is there a specific reason not to do so with these two schemata?

Reply:
I agree, but doing so would have no effect. The difference is that these are response schemas, and so they don't deserialize the passed data, they simply dump it. The data is then assumed to be correct, and no validation occurs regardless of strictness, required fields etc. See [this comment](https://github.com/marshmallow-code/marshmallow/issues/295#issuecomment-146745553).

In my opinion, this could be done differently [in flask-apispec](https://github.com/jmcarp/flask-apispec/blob/master/flask_apispec/wrapper.py#L58) as I think returning a response that doesn't match the marshalling schema should raise an internal validation error.